packages/diagram-ui/src/features/toolbox/toolbox.ts

```diff
@@ -0,0 +1,548 @@
+                this.currentRoot = root;
+                this.initializeToolbox();
+                this.elementPreviews.clear();
+                this.searchIndex = undefined;
+            }
+        } else if (TransactionalAction.isTransactionalAction(action)) {
+            if (action.committed) {
+                this.pointerEventsDisabled = false;
+                this.update();
+            }
+        } else if (EditorConfigUpdatedAction.is(action)) {
+            this.isClosed = action.config.toolboxDisabled;
+            this.update();
+        }
+    }
+
+    /**
+     * Initializes the toolbox.
+     */
+    private initializeToolbox(): void {
+        if (!this.containerElement) {
+            this.initialize();
+        }
+        this.setContainerVisible(true);
+        this.update();
+    }
+
+    /**
+     * Updates the toolbox by re-rendering it.
+     */
+    private update(): void {
+        if (this.currentRoot != undefined) {
+            this.currentVNode = this.patcherProvider.patcher(
+                this.currentVNode!,
+                this.generateToolbox(this.currentRoot)
+            );
+        }
+    }
+
+    override id(): string {
+        return Toolbox.ID;
+    }
+
+    override containerClass(): string {
+        return "toolbox-wrapper";
+    }
+
+    protected override initializeContents(containerElement: HTMLElement): void {
+        const toolbox = document.createElement("div");
+        containerElement.appendChild(toolbox);
+        containerElement.classList.add("hylimo");
+        this.currentVNode = this.patcherProvider.patcher(toolbox, h("div"));
+    }
+
+    /**
+     * Generates the toolbox UI.
+     *
+     * @param root The current root element
+     * @returns The toolbox UI
+     */
+    private generateToolbox(root: Root): VNode {
+        return h(
+            "div.toolbox",
+            {
+                class: {
+                    "pointer-events-disabled": this.pointerEventsDisabled,
+                    closed: this.isClosed
+                }
+            },
+            [
+                h("div.toolbox-header", [h("span.title", "Toolbox"), ...this.generateToolboxButtons()]),
+                this.generateSearchBox(),
+                h("div.items", this.generateToolboxItems(root))
+            ]
+        );
+    }
+
+    /**
+     * Generates the toolbox buttons.
+     *
+     * @returns The toolbox buttons
+     */
+    private generateToolboxButtons(): VNode[] {
+        const buttons: VNode[] = [];
+        if (this.isClosed) {
+            buttons.push(this.generateCodiconButton("tools", "Open Toolbox", () => this.toggleToolbox()));
+        } else {
+            buttons.push(
+                this.generateCodiconButton(
+                    "search",
+                    "Search",
+                    () => {
+                        this.searchString = this.searchString != undefined ? undefined : "";
```

Comment:
Wait, we are deleting the search text after every use?
I mean, both "keeping the search text" and "deleting it after the search" have their advantage…

Reply:
We delete it when you close the search bar. Searching does not close the search bar automatically

Reply:
Am I misreading something here?
After all, this code looks like `when we click on "search", unset the search text`…

Reply:
![image](https://github.com/user-attachments/assets/cd0e9ff2-b2dc-4ed3-b715-910fc1333b41)
it's that button
that toggles the search bar

Reply:
Ah, I see.
I thought that button was something like `Enter`.
The search is run automatically as I hoped for, so deleting the text here is alright.